ceph_deploy/mon.py

```diff
@@ -535,6 +528,26 @@ def make(parser):
 # Helpers
 #
 
+def get_mon_member(args, init_tags=False):
+    """
+    Abstracte the process about getting mon_member.
+
+    @init_tags for choose normal/init stage
+
+    return value are monitor members
+    """
+    cfg = conf.ceph.load(args)
+    if init_tags is not True:
+        mon_initial_members = cfg.safe_get('global', 'mon_initial_members')
+        monitors = re.split(r'[,\s]+', mon_initial_members)
+        return monitors
+    else:
+        cfg_initial_members = cfg.safe_get('global', 'mon_initial_members')
+        if cfg_initial_members is None:
+            raise RuntimeError('No `mon initial members` defined in config')
+        mon_initial_members = re.split(r'[,\s]+', cfg_initial_members)
+        return mon_initial_members
+
```

Comment:
I'm a bit confused by the structure of this.  It seems the only difference between the two code paths (whether or not init_tags is True) is whether or not it raises an exception if cfg.safe_get() returns None.  This seems much easier written as:

```python
def get_mon_initial_members(args, error_on_empty=False):
    cfg = conf.ceph.load(args)
    mon_initial_members = cfg.safe_get('global', 'mon_initial_members')
    monitors = re.split(r'[,\s]+', mon_initial_members)
    if error_on_empty:
        raise exc.NeedHostError(
            'could not find `mon initial members` defined in ceph.conf'
         )
    return mon_initial_members
```

After looking closer at this, since this function is only to get mon_initial_members from the config file, let's rename it to get_mon_initial_members()

It would be better to raise exc.NeedHostError() rather than the RuntimeError, since that's what was raised in the other places.